pkg/ingester/ingester.go

```diff
@@ -1928,6 +1906,73 @@ func (i *Ingester) openExistingTSDB(ctx context.Context) error {
 	return nil
 }
 
+func (i *Ingester) getConcurrencyConfig(userCount int) (tsdbOpenConcurrency, walReplayConcurrency int) {
```

Comment:
In my previous comment I was suggesting making it a standalone function (taking in input `i.cfg.BlocksStorageConfig.TSDB`) and without having `*Ingester` as function receiver. The reason was that after you do it, it would be very easy to unit test and I think you should unit test it too :)